src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx

```diff
@@ -0,0 +1,163 @@
+        display: 'flex',
+        alignItems: 'center',
+        maxWidth: '225px',
+    },
+    reloadButton: {
+        cursor: 'pointer',
+    },
+    buttonTitle: {
+        maxWidth: '200px',
+        '& button': {
+            '& span': {
+                fontSize: 'var(--pf-global--FontSize--sm)',
+                color: 'var(--pf-global--primary-color--100)',
+            },
+            '&:hover, &:focus': {
+                '& span': {
+                    color: 'var(--pf-global--primary-color--200)',
+                },
+            },
+        },
+    },
+    icon: {
+        maxWidth: '25px',
+        color: 'var(--pf-global--primary-color--100)',
+    },
+})
+
+export const getPollInterval = (cookieKey: string) => {
+    let pollInterval = DEFAULT_REFRESH_TIME * 1000
+    if (cookieKey) {
+        const savedInterval = localStorage.getItem(cookieKey)
+        if (savedInterval) {
+            try {
+                const saved = JSON.parse(savedInterval)
+                if (saved.pollInterval !== undefined) {
+                    pollInterval = saved.pollInterval
+                }
+            } catch (e) {
+                //
+            }
+        } else {
+            savePollInterval(cookieKey, pollInterval)
+        }
+    }
+    return pollInterval
+}
+
+export const savePollInterval = (cookieKey: string, pollInterval: number) => {
+    localStorage.setItem(cookieKey, JSON.stringify({ pollInterval }))
+}
+
+export function AcmAutoRefreshSelect(props: AcmAutoRefreshSelectProps) {
+    const [isOpen, setOpen] = useState<boolean>(false)
+    const [selected, setSelected] = useState<RefreshOption>({
+        id: 'refresh-30s',
+        text: 'Refresh every 30s',
+        pollInterval: 30,
+    })
+    const [pollInterval, setPollInterval] = useState(props.pollInterval)
+    const classes = useStyles()
+    const { refetch, refreshValues = [] } = props
+
+    const autoRefreshChoices = refreshValues.map((pollInterval) => {
+        let id, text
+        if (pollInterval >= 60) {
+            id = `refresh-${pollInterval / 60}m`
+            text = `Refresh every ${pollInterval / 60}m`
+        } else if (pollInterval !== 0) {
+            id = `refresh-${pollInterval}s`
+            text = `Refresh every ${pollInterval}s`
+        } else {
+            id = 'refresh-disable'
+            text = 'Disable refresh'
+        }
+        pollInterval *= 1000
+        return { id, text, pollInterval }
+    })
+
+    const handleOptionClick = (item: RefreshOption) => {
+        setSelected(item)
+        setPollInterval(item.pollInterval)
+    }
+
+    useEffect(() => {
+        console.log('selected', selected)
+        console.log('pollInterval', pollInterval)
+
+        const { refreshCookie, startPolling, stopPolling } = props
+        if (pollInterval === 0) {
+            stopPolling()
+        } else {
+            startPolling(pollInterval)
+        }
+        savePollInterval(refreshCookie, pollInterval)
+    }, [selected])
```

Comment:
might want to include a return here which is triggered on dismount?

Reply:
Opting to replace start/stop polling with refetch on interval.  But you are correct.